### ISSUE TITLE
[BUG] What is utils::path-cache::getAAP() doing?

### ISSUE THREAD

**Issue by szatanjl** [justkey007/tsc-alias#155]
```typescript
    const aliasPathParts = aliasPath
      .split('/')
      .filter((part) => !part.match(/^\.$|^\s*$/));

    let aliasPathPart = aliasPathParts.shift() || '';

    let pathExists: boolean;
    while (
      !(pathExists = this.exists(join(basePath, aliasPathPart))) &&
      aliasPathParts.length
    ) {
      aliasPathPart = aliasPathParts.shift();
    }
```

This is the code in question.  What is this while loop doing here?  Makes no sense to me.

I have following configured in tsconfig

```json
"compilerOptions": {
"paths": {
    "@asdf": [ "./asdf1.json", "./asdf2.json" ] 
},
"baseUrl": "src",
```

And in my case "asdf1.json" is missing, "asdf2.json" is present.

So the default.replacer.js goes over these in order, calculates the absolute alias path (using getAAP()) and checks if it exists.
If not then goes to option no 2 and so on.  This would be great if not for the fact that getAAP() is broken and so it fails.

What happens inside getAAP():

1. basePath = "/home/.../project/dist"; aliasPath = "./asdf1.json"
2. convert aliasPath into aliasPathParts
3. aliasPathParts = [ "asdf1.json" ]
4. enter while loop:
5. check if "/home/.../project/dist/asdf1.json" exists
6. it doesn't so quit loop
8. return "/home/.../project/dist"

It removed the "version.json" suffix from the path.  And then default replacer checks (again) if the absoluteAliasPath exists and it does (because it removed the "version.json") and so default replacer happily changes "@asdf" to "." instead of seeing that file does not exist and going to the second one.

TL;DR getAAP() returns wrong results.  join(x, y) does not do x += y
